src/v/archival/upload_housekeeping_service.h

```diff
@@ -92,8 +92,13 @@ class housekeeping_workflow {
     /// Housekeeping state
     housekeeping_state state() const;
 
-    /// Returns true if one of the jobs is executed right now
-    bool has_active_job() const;
+    /// Returns the name of the job currently executing or nullopt if no job is
+    /// currently running.
+    std::optional<ss::sstring> get_active_job_name() const;
+
+    /// Returns a future which resolves when the active job completes. If
+    /// there's no such job, retuns a nullopt.
+    std::optional<ss::future<>> subscribe_to_active_job();
```

Comment:
IIRC we can have multiple concurrent jobs. I don't quite remember for sure.

Reply:
I don't think so. Should be just one job. In any case, I've removed this code.